app/assets/javascripts/dashboard/views/dataset_view.js

```diff
@@ -86,6 +86,19 @@ class DatasetView extends React.PureComponent<Props, State> {
     this.setState({ searchQuery: event.target.value });
   };
 
+  updateDatasets = (newDataset: DatasetType) => {
+    const newDatasets = this.state.datasets.map((dataset: DatasetType) => {
+      if (dataset.name === newDataset.name) {
+        return Object.assign({}, newDataset);
```

Comment:
You already copied the dataset on the caller side of this function, which is why I'd just write `return newDataset`.